web/app/js/components/ResourceDetail.jsx

```diff
@@ -149,9 +150,14 @@ export class ResourceDetailBase extends React.Component {
         // downstream resources of this resource (meshed traffic only)
         this.api.fetchMetrics(
           `${this.api.urlsForResource("all")}&from_name=${resource.name}&from_type=${resource.type}&from_namespace=${resource.namespace}`
-        )
+        ),
       ];
 
+    if (!this.state.queryForDefinition) {
```

Comment:
I think this should check if `queryForDefinition` equals `true`. If it's `false` we don't need to fetch the definition. Same with line 178.